File: pkg/utils/logger.go
package utils

import (
	"fmt"
	"github.com/fatih/color"
)

// Logger provides colored logging methods
var Logger = struct {
	Info  func(string)
	Ready func(string)
	Warn  func(string)
	Error func(string)
	Event func(string)
}{
	Info: func(msg string) {
		fmt.Println(color.BlueString("•") + " " + color.HiBlackString("info  - ") + msg)
	},
	Ready: func(msg string) {
		fmt.Println(color.GreenString("•") + " " + color.HiBlackString("ready - ") + msg)
	},
	Warn: func(msg string) {
		fmt.Println(color.YellowString("•") + " " + color.HiBlackString("warn  - ") + msg)
	},
	Error: func(msg string) {
		fmt.Println(color.RedString("•") + " " + color.HiBlackString("error - ") + msg)
	},
	Event: func(msg string) {
		fmt.Println(color.CyanString("•") + " " + color.HiBlackString("event - ") + msg)
	},
}

File: src/utils/logger.js
import Color from './color.js'

const getTimestamp = () => {
    return Color.dim(`[${new Date().toLocaleTimeString('id-ID', { hour12: false })}]`)
}

const logger = {
    info: (msg) => console.log(`${getTimestamp()} ${Color.blue('INFO')} ${msg}`),
    done: (msg) => console.log(`${getTimestamp()} ${Color.green('DONE')} ${msg}`),
    warn: (msg) => console.log(`${getTimestamp()} ${Color.yellow('WARN')} ${msg}`),
    error: (msg) => console.log(`${getTimestamp()} ${Color.red('ERR')} ${msg}`),
    ready: (msg) => console.log(`${getTimestamp()} ${Color.cyan('READY')} ${msg}`),
    config: (msg) => console.log(`${getTimestamp()} ${Color.magenta('CONF')} ${msg}`)
}

export default logger
